### PR TITLE
chore: various linting config changes

### PR DESCRIPTION
**Pull request by smarshall-spitzbart**
# Description

- Ensures linting for both the GH workflow and makefile reference the same config file, `.golangci.yml`
- `legacy_ibc_testing` and entire `testutil` folder are always excluded from linting
- Introduces new makefile targets, `make lint` runs the linter without changing code, `make format` formats the code 

## Linked issues

Not aware of any linked issues

## Type of change

If you've checked more than one of the first three boxes, consider splitting this PR into multiple PRs!

- [ ] `Feature`: Changes and/or adds code behavior, irrelevant to bug fixes
- [ ] `Fix`: Changes and/or adds code behavior, specifically to fix a bug
- [ ] `Refactor`: Changes existing code style, naming, structure, etc.
- [ ] `Testing`: Adds testing
- [ ] `Docs`: Adds documentation
- [x] Misc linting shiz + makefile changes

## Regression tests

n/a

## New behavior tests

n/a

## Versioning Implications

- [ ] This PR will affect [semantic versioning as defined for ICS](../CONTRIBUTING.md#semantic-versioning)

If the above box is checked, which version should be bumped?

- [ ] `MAJOR`: Consensus breaking changes to both the provider and consumers(s), including updates/breaking changes to IBC communication between provider and consumer(s)
- [ ] `MINOR`: Consensus breaking changes which affect either only the provider or only the consumer(s)
- [ ] `PATCH`: Non consensus breaking changes

## Targeting

Please select one of the following:

- [x] This PR is only relevant to main
- [ ] This PR is relevant to main, and should also be back-ported to ____ (ex: v1.0.0 and v1.1.0)
- [ ] This PR is only relevant to ____ (ex: v1.0.0, v1.1.0, and v1.2.0)

### CODE REVIEW

File: .github/workflows/golangci-lint.yml [hunk 1]
```diff
@@ -30,7 +30,7 @@ jobs:
           # working-directory: somedir
 
           # Optional: golangci-lint command line arguments.
-          args: --timeout=20m --skip-dirs legacy_ibc_testing
```
Comment (smarshall-spitzbart) [inline .github/workflows/golangci-lint.yml:33]: These args are set from `.golangci.yml`

File: Makefile [hunk 2]
```diff
@@ -67,6 +67,19 @@ test-gaia-e2e-parallel-tagged:
 test-no-cache:
 	go test ./... -count=1 && go run ./tests/e2e/...
 
+###############################################################################
+###                                Linting                                  ###
+###############################################################################
+
+lint:
+	@echo "--> Running linter"
+	golangci-lint run  ./... --config .golangci.yml
```
Comment (mpoke) [inline Makefile:76]: Should make install a particular version of golangci-lint first? e.g.,  https://github.com/cosmos/cosmos-sdk/blob/58f3a4a2375f0b617ee0ac3399085c1f996195ba/Makefile#L387

Comment (smarshall-spitzbart) [reply Makefile:79]: Yea I've added that here https://github.com/cosmos/interchain-security/pull/860/commits/032ce34f5096974c402b49e2b046e24b5c08ba3d, the version in your link didn't build for me, but I think it's that I still have go 1.18 installed. Might be worth waiting on #840

**Review by MSalopek — APPROVED** [review]
Approved!

**Comment by smarshall-spitzbart** [comment]
`make test-e2e-short` passes locally for me

**Review by mpoke — APPROVED** [review]
LGTM

**Pull request merged**